library/src/main/java/com/novoda/downloadmanager/lib/DownloadService.java

```diff
@@ -419,9 +419,12 @@ private void deleteFileAndMediaReference(DownloadInfo info) {
 
     private boolean kickOffDownloadTaskIfReady(boolean isActive, DownloadInfo info) {
```

Comment:
The confusion and strangeness comes from the fact that the `isActive` is "accumulated" by passing it, applying an `|` and then returning it, and at the next pass in the while loop (https://github.com/novoda/download-manager/blob/Fix-bug-wrong-download-status-when-is-not-ready-to-download/library/src/main/java/com/novoda/downloadmanager/lib/DownloadService.java#L330) the method will be executed with the accumulated value up until now.